packages/cozy-mespapiers-lib/src/utils/search.js

```diff
@@ -0,0 +1,120 @@
+import { Document } from 'flexsearch'
+
+import { models } from 'cozy-client'
+
+import { CONTACTS_DOCTYPE } from '../doctypes'
+
+const {
+  document: {
+    themes: { themesList }
+  },
+  file: { isFile }
+} = models
+
+export const index = new Document({
+  document: {
+    id: '_id',
+    tag: 'flexsearchProps:tag',
+    index: [
+      'flexsearchProps:name',
+      'flexsearchProps:translatedQualificationLabel',
+      'flexsearchProps:fullname',
+      'flexsearchProps:birthday',
+      'flexsearchProps:birthcity',
+      'flexsearchProps:email',
+      'flexsearchProps:civility',
+      'flexsearchProps:company',
+      'flexsearchProps:jobTitle'
+    ],
+    store: ['_id', '_type']
```

Comment:
By doing this, you tell flexsearch to only keep in memory the `_id` and `_type` fields. This is enough if we want to take advantage of the cozy-client store, to get the document from memory once it is needed (the ideal situation), but I'm not sure we want to do this as a first step?  
Then, you need to pass all the fields that will be useful to display, or just pass `store: true` to store all the fields.

Reply:
A ce stade je ne sais pas quelle est la meilleure approche. Je me suis posé la question, et j'ai lu sur la doc `Do not use a store when: 1. an array of IDs as the result is good enough, or 2. you already have the contents/documents stored elsewhere (outside the index).` https://github.com/nextapps-de/flexsearch#document-store donc je suis parti là-dessus pour l'instant.

Reply:
La version simple (1) c'est un `store: true`, qui va garder tout le doc en mémoire. Donc quand tu fais une recherche, il te retourne simplement tout le doc. L'inconvénient c'est que tout est gardé en mémoire, alors que tu as déjà tout ça dans le store cozy-client, vu qu'on a préalablement requêté tous les papiers/contacts.
La meilleure approche (2) serait donc d'utiliser d'exploiter ça pour ne stocker que l'id / type par flexsearch, et de requêter le store de cozy-client avec ça pour récupérer le doc en entier, en s'assurant que le realtime mette bien le store cozy-client à jour en plus de l'index flexsearch.
Là tu es parti sur (2), mais je ne suis pas sûr que c'était ce que tu voulais en réalité.